drivers/flash/flash_stm32h7x.c

```diff
@@ -64,6 +64,67 @@ struct flash_stm32_sector_t {
 	volatile uint32_t *sr;
 };
 
+#if defined(CONFIG_FLASH_STM32_READOUT_PROTECTION)
+uint8_t flash_stm32_get_rdp_level(const struct device *dev)
+{
+	FLASH_TypeDef *regs = FLASH_STM32_REGS(dev);
+
+	return (regs->OPTSR_CUR & FLASH_OPTSR_RDP_Msk) >> FLASH_OPTSR_RDP_Pos;
+}
+
+void flash_stm32_set_rdp_level(const struct device *dev, uint8_t level)
```

Comment:
Could you implement `write_optb()` function similar to what we have in `flash_stm32f4.c` and use it in this function? We plan to add write protection later.

Reply:
The code is already duplicated. It doesn't matter if it's in `flash_stm32_set_rdp_level()` in `flash_stm32h7x.c` or in `write_optb()`. As I mentioned earlier we plan to add support for write protection in near future.

Reply:
I see that you added `write_optb()` in the next commit. Can we squash `drivers: flash: stm32h7 flash has a write_optb function to set rdp level` with this commit?